tests/phpunit/integration/Modules/Analytics_4Test.php

```diff
@@ -147,6 +147,47 @@ public function test_register() {
 		);
 	}
 
+	public function test_register__replicate_analytics_sharing_settings() {
```

Comment:
It looks like this test should check one more case, which is the scenario where both `analytics` and `analytics-4` have settings, in which case the replication should also not occur.

As the method is starting to get a bit long with multiple scenarios being tested within it, it might also be worth a refactor to make use of a `@dataProvider`...